### DISCUSSION TITLE
Use the image without VPN

### DISCUSSION THREAD

**Discussion by Telestosatt** [haugene/docker-transmission-openvpn#2479] [General]
Hello
Can we use this image without vpn?
I mean.... use this image to run transmission via docker without vpn?

Yes, I know the sole purpose of this image is to use transmission over vpn. 
But
Is it possible to add this feature? via some env like `-e VPN_ENABLED=<yes|no>`
In that way, this image will be useful to more users.

[Deluge over vpn](https://hub.docker.com/r/binhex/arch-delugevpn) image has [this feature](https://i.imgur.com/KmcgGp6.png). 
Would like to have the same for our beloved transmission as well :)

**Comment by ilike2burnthing** [comment]
https://github.com/haugene/docker-transmission-openvpn/discussions/2143

If you just want to use Transmission - https://hub.docker.com/r/linuxserver/transmission

**Comment by Telestosatt** [comment]
Thanks for the reply.
I had came around this image while I was searching.  Yet to try it out.
I am using the current image for some time. I had made some very minor tweaks and made my own docker image. I was hoping to add `vpn as optional` feature to the current image.

I guess ... I need to look into linked image and have my way around it.
Thanks again.

**Comment by eUnisk** [comment]
I will need this feature actually. You may want to use transmission without a VPN for a while.  Reasons are for testing, or while your VPN is not working, or just stopping using VPN but you don't want to reconfigure another image all over again.

**Comment by edgd1er** [comment]
At the moment, the way the container is running, it is not possible easily
Openvpn is launched at boot time,
https://github.com/haugene/docker-transmission-openvpn/blob/51e5f1bcbbbc3c6c54b09896a44b6165531a1305/Dockerfile#L92
then openvpn execute tunnelUp script upon connection.
https://github.com/haugene/docker-transmission-openvpn/blob/51e5f1bcbbbc3c6c54b09896a44b6165531a1305/openvpn/start.sh#L216

There are two points that are important, I think:
* Container startup process logic will have to be revised to check if a vpn connection is required or not. healthcheck will have to be adjusted.
* the main security here is when the openvpn connection is lost, the container stops. Introducing a YES/NO flag for VPN, you could have a container running transmission, without a valid vpn connection. The overhead to check all use cases and ensure vpn is running when required is not that light. 

I guess a container with services (S6-overlay or supervisor ) would be a good start.
Not sure though the maintainers wish to take that path.

**Reply by ksurl** [reply]
the easy way is to use the lsio image. but you could if you really want to retain all the settings and other setup, override the entrypoint to launch transmission directly.
